### DISCUSSION TITLE
Please sort out warnings at least all level 3 and most Level 4

### DISCUSSION THREAD

**Discussion by Smurf-IV** [pocoproject/poco#4149] [Ideas]
e.g. in Foundation using VS170 project with no adjustments (i.e. still \W3) the following are seen:
``` c++
Severity	Code	Description	Project	File	Line
Warning	C4244	'=': conversion from 'uint64_t' to 'poco_double_conversion::Bignum::Chunk', possible loss of data (compiling source file src\NumericString.cpp)	Foundation	Z:\Repos\SRLE\Thirdparty\PoC\PocoProject.org\Foundation\src\bignum.cc	149
Warning	C4267	'initializing': conversion from 'size_t' to 'int', possible loss of data	Foundation	Z:\Repos\SRLE\Thirdparty\PoC\PocoProject.org\Foundation\src\RegularExpression.cpp	326
Warning	C4267	'initializing': conversion from 'size_t' to 'int', possible loss of data	Foundation	Z:\Repos\SRLE\Thirdparty\PoC\PocoProject.org\Foundation\src\RegularExpression.cpp	327
```

On Level 4 (The company requirement and also to have the them as errors ....)
Then there is a lot of noise coming through:
![image](https://github.com/pocoproject/poco/assets/2418812/ffb5edb4-6efc-4b96-b10c-11a92ca0536a)

**Comment by aleks-f** [comment]
Please send a pull request with fixes to devel branch. Narrowing warnings are fixable with static_cast. The double_conversion warnings are upstream and won't be fixed in poco. Upgrading double conversion to latest release may help.

**Comment by andrewauclair** [comment]
If you're using VS2022 then you can just set external includes to W0. Also, if you're using W4 you surely have a list of ignored warnings, right? C4100 is basically just noise that can be ignored, along with several other warnings.
